packages/wrangler/src/dev.tsx

```diff
@@ -626,12 +626,16 @@ export async function startApiDev(args: StartDevOptions) {
 	};
 }
 /**
+ * Get an available TCP port number.
+ *
  * Avoiding calling `getPort()` multiple times by memoizing the first result.
  */
-function memoizeGetPort(defaultPort?: number) {
+function memoizeGetPort(defaultPort: number, host: string) {
 	let portValue: number;
 	return async () => {
-		return portValue || (portValue = await getPort({ port: defaultPort }));
+		// Check a specific host to avoid probing all local addresses.
+		portValue = portValue ?? (await getPort({ port: defaultPort, host: host }));
```

Comment:
NIT - this can also be:

```suggestion
		portValue = portValue ?? (await getPort({ port: defaultPort, host }));
```